src/CVRF/cvrf_enumeration.c

```diff
@@ -0,0 +1,178 @@
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <string.h>
+
+#include <libxml/xmlreader.h>
+#include <libxml/xmlwriter.h>
+
+#include "public/cvrf.h"
+#include "cvrf_priv.h"
+
+#include "common/list.h"
+#include "common/_error.h"
+#include "common/xmltext_priv.h"
+#include "common/elements.h"
+#include "common/oscap_string.h"
+#include "common/util.h"
+
+#include "source/oscap_source_priv.h"
+#include "source/public/oscap_source.h"
+
+
+/*****************************************************************************
+ * Static functions
+ */
+
+static int cvrf_enumeration_attr(xmlTextReaderPtr reader, char *attname, const struct oscap_string_map *map) {
+	char *attrstr = (char *)xmlTextReaderGetAttribute(reader, BAD_CAST attname);
+	if (attrstr == NULL)
+		return 0;
+	int ret = oscap_string_to_enum(map, attrstr);
+	oscap_free(attrstr);
+	return ret;
+}
+
+static int cvrf_enumeration_node_value(xmlTextReaderPtr reader, const struct oscap_string_map *map) {
+	char *valuestr = (char *)oscap_element_string_copy(reader);
+	if (valuestr == NULL)
+		return 0;
+	int ret = oscap_string_to_enum(map, valuestr);
+	oscap_free(valuestr);
+	return ret;
+}
+
+static const char *cvrf_enumeration_get_text(const struct oscap_string_map *map, int val) {
+	return oscap_enum_to_string(map, val);
+}
+static int cvrf_enumeration_from_text(const struct oscap_string_map *map, const char *text) {
+	return oscap_string_to_enum(map, text);
+}
+
+
+/*****************************************************************************
+ * CVRF enum map definitions
+ */
+
+const struct oscap_string_map CVRF_DOC_PUBLISHER_TYPE_MAP[] = {
+	{CVRF_DOC_PUBLISHER_VENDOR, "Vendor"},
+	{CVRF_DOC_PUBLISHER_DISCOVERER, "Discoverer"},
+	{CVRF_DOC_PUBLISHER_COORDINATOR, "Coordinator"},
+	{CVRF_DOC_PUBLISHER_USER, "User"},
+	{CVRF_DOC_PUBLISHER_OTHER, "Other"},
+	{0, NULL}
+};
+
+cvrf_doc_publisher_type_t cvrf_doc_publisher_type_parse(xmlTextReaderPtr reader, char *attname) {
+	return cvrf_enumeration_attr(reader, attname, CVRF_DOC_PUBLISHER_TYPE_MAP);
+}
+const char *cvrf_doc_publisher_type_get_text(cvrf_doc_publisher_type_t doc_publisher_type) {
+	return cvrf_enumeration_get_text(CVRF_DOC_PUBLISHER_TYPE_MAP, doc_publisher_type);
+}
+
+
+const struct oscap_string_map CVRF_DOC_STATUS_TYPE_MAP[] = {
+	{CVRF_DOC_STATUS_DRAFT, "Draft"},
+	{CVRF_DOC_STATUS_INTERIM, "Interim"},
+	{CVRF_DOC_STATUS_FINAL, "Final"},
+	{0, NULL}
+};
+
+cvrf_doc_status_type_t cvrf_doc_status_type_parse(xmlTextReaderPtr reader) {
```

Comment:
does this need to be a separate function? why is the function not static, is it part of the API?